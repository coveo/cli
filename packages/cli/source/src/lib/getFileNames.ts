import {CliUx} from '@oclif/core';

export interface FilesOrFolders {
  file?: string[]; // TODO: CDX-856: remove flag
  folder?: string[]; // TODO: CDX-856: remove flag
  files?: string[];
}

export function getFileNames(entries: FilesOrFolders, commandName: string) {
  // TODO: CDX-856: only read --files flag
  const entryNames = [
    ...(entries.file ?? []),
    ...(entries.folder ?? []),
    ...(entries.files ?? []),
  ];

  if (entryNames.length === 0) {
    CliUx.ux.error(
      `You must set the \`files\` flag. Use \`${commandName} --help\` to get more information.`
    );
  }

  return entryNames;
}
