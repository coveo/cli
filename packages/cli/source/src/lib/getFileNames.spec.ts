import {getFileNames} from './getFileNames';
import {stderr} from 'stdout-stderr';

describe('getFileNames()', () => {
  beforeEach(() => {
    stderr.start();
  });
  afterEach(() => {
    stderr.stop();
  });

  it('should throw with the command when no entries is provided', () => {
    expect(() => {
      getFileNames({files: []}, 'holdthedoor.bat');
    }).toThrowErrorMatchingSnapshot();
  });

  it('should concat the entries provided', () => {
    expect(
      getFileNames(
        {files: ['foo', 'bar'], folder: ['fizz', 'buzz'], file: ['baz']},
        'holdthedoor.bat'
      )
    ).toEqual(['baz', 'fizz', 'buzz', 'foo', 'bar']);
  });
});
