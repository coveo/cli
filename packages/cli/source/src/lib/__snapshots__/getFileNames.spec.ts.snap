// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFileNames() should throw with the command when no entries is provided 1`] = `"You must set the \`files\` flag. Use \`holdthedoor.bat --help\` to get more information."`;
